File: UTTrax/src/main/scala/Game/TestPlayer/Player.scala
package Game.TestPlayer

import Game.World._
import Game.World.traxColor.traxColor

//import TraxWorld._

/**
 * Created by proska on 6/29/15.
 */
trait Player {

  val side:traxColor
  var state = new gameState

  def initialize(): Unit ={
    var tmp = new Route
    tmp.start = (Coordinate(0,1),Margin.DOWN)
    tmp.end   = (Coordinate(0,-1),Margin.TOP)
    tmp.length = 1
    state.whiteRoutes = List(tmp)

    var tmp1 = new Route
    tmp1.start = (Coordinate(1,0),Margin.LEFT)
    tmp1.end   = (Coordinate(-1,0),Margin.RIGHT)
    tmp1.length = 1
    state.blackRoutes = List(tmp1)
  }

  def play(): Move

  def update(move: Move):Any = {
    state.updateState(move,side)
  }
}

File: UTTrax/src/main/scala/Game/TestPlayer/Player.java
package Game.TestPlayer;

import Game.World.*;
import scala.Boolean;


/**
* Created by proska on 8/1/15.
*/
public interface Player {

    traxColor side = null;

//    gameState state = new gameState();

    public void initialize();

    public Move play();

    public void update(Move move);

    public gameState getState();
    public void setState(gameState st);
}

